src/interface/swarm_comms.cpp

```diff
@@ -443,18 +356,37 @@ void Swarm::UnloadBuffers_() {
     int realPackDim = vreal.GetDim(2);
     int intPackDim = vint.GetDim(2);
 
-    // construct map from buffer index to swarm index (or just return vector of
-    // indices!)
     const int particle_size = GetParticleDataSize();
     auto swarm_d = GetDeviceContext();
 
+    // Change meaning of neighbor_received_particles from particles per neighbor to
+    // cumulative particles per neighbor
+    int val_prev = 0;
+    for (int n = 0; n < nbmax; n++) {
+      double val_curr = neighbor_received_particles_h(n);
```

Comment:
This is an `int` isn't it?
```suggestion
      auto val_curr = neighbor_received_particles_h(n);
```

Reply:
Good catch! Scary that that just compiled without warnings and worked. Warning/erroring on implicit casts would probably cause a whole bunch of headaches though. Anyway, this is fixed now.